pkg/networkmanager/interfaces.js

```diff
@@ -3165,6 +3510,134 @@ function connection_devices(con) {
     return devices;
 }
 
+PageNetworkGeneralSettings.prototype = {
+    _init: function () {
+        this.id = "network-general-settings-dialog";
+        this.general_settings_template = $("#network-general-settings-template").html();
+        mustache.parse(this.general_settings_template);
+    },
+
+    setup: function () {
+        $('#network-general-settings-cancel').click($.proxy(this, "cancel"));
+        $('#network-general-settings-apply').click($.proxy(this, "apply"));
+    },
+
+    enter: function () {
+        $('#network-general-settings-error').hide();
+        this.settings = PageNetworkGeneralSettings.ghost_settings || PageNetworkGeneralSettings.connection.copy_settings();
+        this.update();
+    },
+
+    show: function() {
+    },
+
+    leave: function() {
+    },
+
+    update: function() {
+        const self = this;
+        const options = self.settings.connection;
+        const model = PageNetworkGeneralSettings.model;
+        const setts = model.get_settings().Connections;
+
+        function vpn_connections_handler() {
```

Comment:
This is indeed related to selecting a connection's secondary connection (A VPN from the system).
Now I see this would be better off in a different PR since it's not strictly related to the wifi manager but rather to  networkmanager as a whole. Is it worth splitting?

Reply:
I think so. Or at least separate commit in this PR.
(and yes, now here are 13 commits, but should be two (or 3 with this vpn))